hydromt/raster.py

```diff
@@ -2962,33 +3050,22 @@ def to_mapstack(
             raise ValueError(f"Extension unknown for driver: {driver}")
         ext = gis_utils.GDAL_EXT_CODE_MAP.get(driver)
         os.makedirs(root, exist_ok=True)
-        with tempfile.TemporaryDirectory() as tmpdir:
-            if driver == "PCRaster" and _compat.HAS_PCRASTER:
-                clone_path = gis_utils.write_clone(
-                    tmpdir,
-                    gdal_transform=self.transform.to_gdal(),
-                    wkt_projection=None if self.crs is None else self.crs.to_wkt(),
-                    shape=self.shape,
-                )
-                profile_kwargs.update({"clone_path": clone_path})
-            for var in self.vars:
-                if "/" in var:
-                    # variables with in subfolders
-                    folders = "/".join(var.split("/")[:-1])
-                    os.makedirs(join(root, folders), exist_ok=True)
-                    var0 = var.split("/")[-1]
-                    raster_path = join(root, folders, f"{prefix}{var0}{postfix}.{ext}")
-                else:
-                    raster_path = join(root, f"{prefix}{var}{postfix}.{ext}")
-                if driver == "PCRaster":
-                    profile_kwargs.update({"pcr_vs": pcr_vs_map.get(var, "scalar")})
-                self._obj[var].raster.to_raster(
-                    raster_path,
-                    driver=driver,
-                    dtype=dtype,
-                    tags=tags,
-                    windowed=windowed,
-                    mask=mask,
-                    logger=logger,
-                    **profile_kwargs,
-                )
+        for var in self.vars:
+            if "/" in var:
+                # variables with in subfolders
+                folders = "/".join(var.split("/")[:-1])
+                os.makedirs(join(root, folders), exist_ok=True)
+                var0 = var.split("/")[-1]
```

Comment:
If you're doing path operations here this is probably going to break on other oses that have different path seperators. Please use the os.path functions for these operations